### PR TITLE
mpi/info: Fix preprocessor guards for error checking

### PR DESCRIPTION
**Pull request by raffenet**
## Pull Request Description

We always check the return code from MPIR_Info_set_impl, so the
failure and exit labels must always be defined. Fixes build error with
--disable-error-checking introduced in [f0bc24757e4b].

## Expected Impact

Fix --disable-error-checking builds.

## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [x] Remove xfail from the test suite when fixing a test
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Passes whitespace checkers
* [x] Passes warning tests
* [ ] Passes all tests
* [x] Add comments such that someone without knowledge of the code could understand
* [x] Add Devel Docs in the `doc/` directory for any new code design

### CODE REVIEW

**Comment by raffenet** [comment]
test:jenkins/ch4/most

**Comment by raffenet** [comment]
test:jenkins/ch4/ucx

File: src/mpi/info/info_set.c [hunk 1]
```diff
@@ -125,7 +122,6 @@ int MPI_Info_set(MPI_Info info, const char *key, const char *value)
     mpi_errno = MPIR_Err_return_comm(NULL, __func__, mpi_errno);
     goto fn_exit;
     /* --END ERROR HANDLING-- */
-#endif
 }
```
Comment (gcongiu) [inline src/mpi/info/info_set.c]: Just for my better understanding ... Why is the way we handle errors here different from other parts of the code? For example, in https://github.com/pmodels/mpich/blob/f73ecd6f12558d241eecf6e3986abb14d7cf2b29/src/mpi/pt2pt/send.c#L156 we don't call `MPIR_Err_create_code` unless we have error checking.

Comment (raffenet) [reply src/mpi/info/info_set.c]: That's a good point. I think the model in send is a better one to copy. I have updated the PR to match it.

**Comment by raffenet** [comment]
test:jenkins/ch4/most

**Comment by raffenet** [comment]
test:mpich/custom --disable-error-checking

**Comment by raffenet** [comment]
> The PR itself looks fine to me.
> 
> Just quickly checked the other functions' error handling. It looks like only `info_get`'s code is inconsistent with the others. It calls `MPIR_Err_return_comm` within the `ifdef HAVE_ERROR_CHECKING` scope (see below). Can we also fix it in this PR?
> 
> ```
>   fn_fail:
> #ifdef HAVE_ERROR_CHECKING
> ...
>     mpi_errno = MPIR_Err_return_comm(NULL, FCNAME, mpi_errno);
> #endif
> ```

Sure. New version coming shortly.

**Comment by raffenet** [comment]
test:jenkins/ch4/most
test:mpich/custom --disable-error-checking

**Comment by raffenet** [comment]
Not sure what happened with Jenkins tests. Looks to be system software related build errors, none of which this PR should touch. Re-testing...
test:jenkins/ch4/ofi
test:mpich/custom --disable-error-checking

**Pull request merged**